src/client/content_cao.cpp

```diff
@@ -1432,6 +1448,17 @@ void GenericCAO::updateBonePosition()
 			bone->updateAbsolutePosition();
 		}
 	}
+	// The following is needed for set_bone_pos to propagate to
+	// attached objects correctly.
+	// Irrlicht ought to do this, but doesn't when using EJUOR_CONTROL.
+	for (u32 i = 0; i < m_animated_meshnode->getJointCount(); ++i) {
+		auto bone = m_animated_meshnode->getJointNode(i);
+		// Look for the root bone.
+		if (bone && bone->getParent() == m_animated_meshnode) {
+			// Update entire skeleton.
+			bone->updateAbsolutePositionOfAllChildren();
+		}
```

Comment:
Can there be more than one root bone or could a `break` statement be added here?

Reply:
Done